lib/pages/trade/usecases/GetSwapInfoUseCase.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:ax_dapp/repositories/SubGraphRepo.dart';
+import 'package:ax_dapp/service/BlockchainModels/SwapInfo.dart';
+import 'package:ax_dapp/service/Controller/usecases/GetPairInfoUseCase.dart';
+import 'package:fpdart/fpdart.dart';
+
+const String _no_swap_info_error_msg = "No swap info found";
+
+class GetSwapInfoUseCase {
```

Comment:
This should be the top level use case that the buy and sell use cases delegate their work to.